### PR TITLE
Preview Code Syntax Highlighting

### PR DESCRIPTION
**Pull request by elocremarc**
Added highlight JS to format and syntax highlight code in the preview. `JS, JSON, CSS and YAML`. Can be used if other code filetypes are added.
eh_s2W5y

### CODE REVIEW

File: static/preview-code.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import url('https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/atom-one-dark.min.css');
```
Comment (raphjaph) [inline static/preview-code.css:1]: Could we get this from jsdelivr as well? At the moment the CSP doesn't allow cdnjs.cloudflare.com

File: static/preview-code.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+html {
+    color: #98a3ad;
+    background-color: #131516;
+    max-width: 900px;
+    font-family: system-ui, sans-serif;
+  }
+
```
Comment (raphjaph) [inline static/preview-code.css:8]: remove last line

File: templates/preview-code.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!doctype html>
+<html lang=en data-inscription={{self.inscription_id}}>  <head>
+    <meta charset=utf-8>
+    <link rel=stylesheet href=/static/preview-code.css>
+    <script src=/static/preview-code.js defer type=module></script>
+  </head>
+  <body>
+  </body>
+</html>
```
Comment (raphjaph) [inline templates/preview-code.html:9]: Think this needs a proper EOL at the end

File: static/preview-code.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import hljs from 'https://cdn.jsdelivr.net/npm/highlightjs@9.16.2/+esm'
```
Comment (raphjaph) [inline static/preview-code.js:1]:
```suggestion
import hljs from 'https://cdn.jsdelivr.net/npm/highlightjs@9/+esm'
```

File: static/preview-code.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+const inscription = document.documentElement.dataset.inscription;
+const response = await fetch(`/content/${inscription}`);
+const code = await response.text();
+const codeBlock = document.createElement('code');
+codeBlock.innerHTML = hljs.highlightAuto(code).value;
```
Comment (raphjaph) [inline static/preview-code.js:7]:
```suggestion
codeBlock.innerHTML = hljs.highlight(code).value;
```
Maybe use only `highlight` and pass in the content type explicitly. You can get the content type from the response object I think

**Review by raphjaph — CHANGES_REQUESTED** [review]
Looks good, other than some small things

File: static/preview-code.js (hunk 4 above)
Comment (elocremarc) [reply static/preview-code.js]: See new imports

File: static/preview-code.js (hunk 5 above)
Comment (elocremarc) [reply static/preview-code.js]: We will have to remember to update the supported languages if we add anymore in `media.rs`. See update

File: templates/preview-code.html (hunk 3 above)
Comment (elocremarc) [reply templates/preview-code.html]: Diffs on github don't seem to show it. I see it in my IDE however.
image

File: static/preview-code.js [hunk 6]
```diff
@@ -1,8 +1,16 @@
...
 
 const inscription = document.documentElement.dataset.inscription;
 const response = await fetch(`/content/${inscription}`);
+const contentType = response.headers.get("content-type");
+const language = contentType.split("/")[1];
 const code = await response.text();
-const codeBlock = document.createElement('code');
-codeBlock.innerHTML = hljs.highlightAuto(code).value;
-document.body.appendChild(codeBlock);
+document.body.innerHTML = `<pre><code>${hljs.highlight(code, {language, ignoreIllegals: true}).value}</code></pre>`;
```
Comment (elocremarc) [inline static/preview-code.js:16]: I missed the `<pre>` which was in the screenshot which does the formatting. Refactored it a bit.

**Comment by elocremarc** [comment]
> Looks good, other than some small things

Should be gtg!

File: static/preview-code.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import hljs from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/highlight.min.js';
+import javascript from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/javascript.min.js';
+import yaml from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/yaml.min.js';
+import css from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/css.min.js';
+import json from 'https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.8.0/build/es/languages/json.min.js';
+
+hljs.registerLanguage('javascript', javascript);
+hljs.registerLanguage('yaml', yaml);
+hljs.registerLanguage('css', css);
+hljs.registerLanguage('json', json);
```
Comment (raphjaph) [inline static/preview-code.js:10]: Is importing and registering the languages necessary?

**Review by raphjaph — APPROVED** [review]
lgtm! 

Bit of a pain that we have to import all languages explicitly but I think it makes sense for now to keep the bundle small.

**Pull request merged**